framework/src/problems/FEProblemBase.C

```diff
@@ -1228,7 +1228,11 @@ FEProblemBase::initialSetup()
   // interfaces. This map will then be used by the AgumentSparsityOnInterface ghosting functor to
   // know which dofs we need ghosted when we call EquationSystems::reinit
   if (_displaced_problem && _mortar_data.hasDisplacedObjects())
+  {
     _displaced_problem->updateMesh();
+    // if displacements were applied to the mesh, the mortar mesh should be updated too
+    updateMortarMesh();
```

Comment:
Someday we might want to enumerate that only the displaced mortar meshes should be updated but since the most common use cases just have mortar running on the displaced mesh, I don't think adding the enumeration is necessarily worth the added code complexity